bldsys/cmake/component_helpers.cmake

```diff
@@ -32,6 +32,8 @@ function(vkb__register_component)
 
         add_library("vkb__${TARGET_NAME}" STATIC ${TARGET_SRC})
 
+        set_property(TARGET "vkb__${TARGET_NAME}" PROPERTY POSITION_INDEPENDENT_CODE ON) # PIC required for shared lib linking
```

Comment:
it is probably better to use set_compile_options("vkb_${TARGET_NAME}" PRIVATE POSITION_INDEPENDENT_CODE ON) 
NB: prior to CMake 3.19, -fPIC didn't mean that pie is also by default mandated in cmake other than in Android builds.  I think we're mandating a recent enough version for this to not be an issue.  Just listing here as a just incase.